File: scala/Task6.scala
object Main extends App {

//O(1) improved solution
  val sm: Int = 100 * (100+1)/ 2
  val squared: Int = (100 * (100 + 1) * (2 * 100 + 1)) / 6

  val result: Int = sm * sm - squared
 
  println(result)

}

File: haskell/Task6.hs
--ghc 7.10

sm :: Int -> Int
squared :: Int -> Int

-- Improved algorithm with O(1) speed
sm til = 
  til * (til+1)/ 2
  
squared til = 
  (til * (til + 1) * (2 * til + 1)) / 6
    
    
main = 
    print $ (sm * sm - squared)
